tests/unit/h/views/oidc_test.py

```diff
@@ -74,158 +86,256 @@ def routes(self, pyramid_config):
         pyramid_config.add_route("oidc.connect.orcid", "/oidc/connect/orcid")
         pyramid_config.add_route("oidc.redirect.orcid", "/oidc/redirect/orcid")
 
+    @pytest.fixture(autouse=True)
+    def secrets(self, secrets):
+        # This just needs to be a string (not a mock) so that it's
+        # JSON-serializable, the tests don't care about the actual value.
+        secrets.token_hex.return_value = "test_rfp"
+        return secrets
+
 
 @pytest.mark.usefixtures("orcid_client_service", "user_service")
 class TestORCIDRedirectViews:
-    def test_when_account_not_yet_connected(
-        self,
-        pyramid_request,
-        orcid_client_service,
-        user_service,
-        user,
-        OAuth2RedirectSchema,
+    @pytest.mark.usefixtures(
+        "with_both_connect_and_login_actions",
+        "assert_no_account_connection_was_added",
+        "assert_user_was_not_logged_in",
+        "assert_no_success_message_was_flashed",
```

Comment:
Hmm... I don't love the pattern of putting asserts in yield fixtures like this. Normally when reading a test you read the description, then the setup, action and finally checks for results and side effects. Here you read the checks first and then the description, setup and action.